### ISSUE TITLE
Printer stops heating before he finished the print

### ISSUE THREAD

**Issue by jensus11** [SoftFever/OrcaSlicer#6839]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.2.0-beta

### Operating System (OS)

macOS

### OS Version

Sequoia 15.0

### Printer

IDEX self build

### How to reproduce

First, I sliced my object and created a gcode file. The Printer printed the gcode very well, but at the half last layer he stopped the heating and printed the rest without heat.
He has in the gcode the line, but I don't know where he got it.
G10 SO ; preheat T-1 time: 30s

![Bildschirmfoto 2024-09-19 um 21 56 40](https://github.com/user-attachments/assets/8d15df84-2144-47ab-bf9f-03060c0f5511)

### Actual results

Bad last layer

### Expected results

  don't stop heating

### Project file & Debug log uploads

![Bildschirmfoto 2024-09-19 um 21 16 39](https://github.com/user-attachments/assets/a1dfdf2c-e45f-4bf9-8c1c-539d96f9ee42)

### Checklist of files to include

- [ ] Log file
- [ ] Project file

**Comment by HaythamB** [comment]
OK did some tracing, and I found this to be probably caused by the M104.1 handling for XL printers, specifically the "export_lines" code block should be moved into the IF statement preceding it to handle cases where the tool index is "-1".

In the below screenshot from the current release (V2.2.0), within GCodeProcessor.cpp in libslic3r>Gcode, lines 4920 to 4967 should be moved to start at line 4919 (ie within the IF code block right before it.

![Screenshot 2024-11-07 111333](https://github.com/user-attachments/assets/381163ec-0455-46b5-af16-c83aad60670b)

**Comment by psychotik2k3** [comment]
is it merged yet ?

**Comment by HaythamB** [comment]
Nope, no action taken yet. I have a windows compiled version with the fix applied, and I've confirmed its working. Guess its just a matter of time.

**Comment by edspeds** [comment]
Any action on this yet?  Don't want to let it close without being addressed.

**Comment by psychotik2k3** [comment]
> Nope, no action taken yet. I have a windows compiled version with the fix applied, and I've confirmed its working. Guess its just a matter of time.

can't you make a PR so it's included in the next release

**Comment by edspeds** [comment]
I believe HaythamB did, I too have his fix and it's working well for me.

**Comment by psychotik2k3** [comment]
nope code is still the same, still not in the if section above.
@HaythamB wana make the PR for you fix please ?

**Comment by HaythamB** [comment]
https://github.com/SoftFever/OrcaSlicer/pull/7405

I did one a long time ago.. Do I need to update something to have it merged?

**Comment by psychotik2k3** [comment]
ok read the Pr and comments it's in workflow.
thanks for the work.
